fe/fe-core/src/main/java/org/apache/doris/load/ExportJob.java

```diff
@@ -264,39 +254,33 @@ public void generateOutfileStatement() throws UserException {
         generateExportJobExecutor();
     }
 
-    public void generateOutfileLogicalPlans(List<String> nameParts)
+    /**
+     * For an ExportJob:
+     * The ExportJob is divided into multiple 'ExportTaskExecutor'
+     * according to the 'parallelism' set by the user.
+     * The tablets which will be exported by this ExportJob are divided into 'parallelism' copies,
+     * and each ExportTaskExecutor is responsible for a list of tablets.
+     * The tablets responsible for an ExportTaskExecutor will be assigned to multiple OutfileStmt
+     * according to the 'TABLETS_NUM_PER_OUTFILE_IN_EXPORT'.
+     *
+     * @throws UserException
+     */
+    public void generateOutfileLogicalPlans(List<String> qualifiedTableName)
             throws UserException {
+        String catalogType = Env.getCurrentEnv().getCatalogMgr().getCatalog(this.tableName.getCtl()).getType();
         exportTable.readLock();
         try {
-            // build source columns
-            List<NamedExpression> selectLists = Lists.newArrayList();
-            if (exportColumns.isEmpty()) {
-                selectLists.add(new UnboundStar(ImmutableList.of()));
-            } else {
-                this.exportColumns.stream().forEach(col -> {
-                    selectLists.add(new UnboundSlot(this.tableName.getTbl(), col));
-                });
-            }
-
-            // get all tablets
-            List<List<Long>> tabletsListPerParallel = splitTablets();
-
-            // Each Outfile clause responsible for MAXIMUM_TABLETS_OF_OUTFILE_IN_EXPORT tablets
-            for (List<Long> tabletsList : tabletsListPerParallel) {
-                List<StatementBase> logicalPlanAdapters = Lists.newArrayList();
-                for (int i = 0; i < tabletsList.size(); i += MAXIMUM_TABLETS_OF_OUTFILE_IN_EXPORT) {
-                    int end = i + MAXIMUM_TABLETS_OF_OUTFILE_IN_EXPORT < tabletsList.size()
-                            ? i + MAXIMUM_TABLETS_OF_OUTFILE_IN_EXPORT : tabletsList.size();
-                    List<Long> tabletIds = new ArrayList<>(tabletsList.subList(i, end));
-
-                    // generate LogicalPlan
-                    LogicalPlan plan = generateOneLogicalPlan(nameParts, tabletIds, selectLists);
-                    // generate  LogicalPlanAdapter
-                    StatementBase statementBase = generateLogicalPlanAdapter(plan);
-
-                    logicalPlanAdapters.add(statementBase);
+            if ("internal".equals(catalogType)) {
+                if (exportTable.getType() == TableType.VIEW) {
+                    // view table
+                    generateViewOrExternalTableOutfile(qualifiedTableName);
+                } else {
```

Comment:
Better use `else if (exportTable.getType() == TableType.OLAP)`,
and add `else` to throw exception for unsupported type.
Because we may add some other type of table later.